cmd/internal/codegen/parse/crd.go

```diff
@@ -492,22 +500,26 @@ func (b *APIs) getMembers(t *types.Type, found sets.String) (map[string]v1beta1.
 
 		// Inline "inline" structs
 		if strat == "inline" {
-			m, r := b.getMembers(member.Type, found)
+			m, r, re:= b.getMembers(member.Type, found)
 			for n, v := range m {
 				members[n] = v
 			}
 			for n, v := range r {
 				result[n] = v
 			}
+			required = append(required, re...)
 		} else {
 			m, r := b.typeToJSONSchemaProps(member.Type, found, member.CommentLines)
 			members[name] = m
 			result[name] = r
+			if !strings.HasSuffix(strat, "omitempty") {
```

Comment:
hmm.... not sure if using `json` tag for this purpose is a good idea ?
have an explicit IDL to specify required field ?
@pwittrock WDYT ?

Reply:
@droot I just recall something. If we specify some field without `omitempty`, openapi generator will generate the openapi definition with this field as required.

Reply:
hmm... if other generators are also looking at "omitempty", then probably it is ok to do and be consistent with that toolset.